### PR TITLE
feat(performance): don't include .map files in perf warnings

### LINKED ISSUE #3469: Add option to exclude files from performance check
**Do you want to request a *feature* or report a *bug*?**
feature

**What is the current behavior?**
webpack throws warning when generated sourcemaps file exceeds max file size.

**What is the expected behavior?**
Add an option to exclude sourcemaps from performance check or exclude them by default.

**If this is a feature request, what is motivation or use case for changing the behavior?**
sourcemaps files would not impact web performance.

**Please mention other relevant information such as the browser version, Node.js version, Operating System and programming language.**
webpack@2.1.0-beta.28

### LINKED ISSUE #3470: Update webpack performance recommendations message to suggest using `import()`
version v2.1.0-beta.28 [release notes](https://github.com/webpack/webpack/releases/tag/v2.1.0-beta.28) suggest using `import()` for  code splitting instead of `System.import`

> add import() as Code Splitting construct. It should be used instead of System.import when possible. System.import will be deprecated in webpack 2 release (removed in webpack 3) as it's behavior is incorrect according to the spec.

 however the performance recommendations message currently suggest using `System.import`

image

### PR DESCRIPTION
**Pull request by TheLarkInn**
**What kind of change does this PR introduce?**
bugfix/feature

**Did you add tests for your changes?**
Yes

**If relevant, link to documentation update:**
N/A

**Summary**
This filters out .map files from the performance budget warnings. In addition I have changed the NoAsyncChunksWarning to use `import()` instead of `System.import` as the messaging since it is now in master and released. 

Fixes #3470 
Fixes #3469 

**Does this PR introduce a breaking change?**
No

### CODE REVIEW

File: lib/Entrypoint.js [hunk 1]
```diff
@@ -41,6 +41,9 @@ Entrypoint.prototype.getSize = function(compilation) {
 	var files = this.getFiles();
 
 	return files
+		.filter(function(asset) {
+			return !(/\.map$/.test(asset))
+		})
```
Comment (sokra) [inline lib/Entrypoint.js]: Maybe it makes sense to remove the Entrypoint.prototype.getSize method altogether and do it somewhere else (in the plugin). It has a weird Compilation dependency and now also a SourceMap dependency...

Comment (TheLarkInn) [inline lib/Entrypoint.js]: Does asset have a Class yet? Maybe this is where this stuff should end up as a refactor point.

**Comment by TheLarkInn** [comment]
@sokra I decided to extract it out like you recommended and instead stuck it inside of SizeFormatHelpers. A better centralized location for both places it was needed.

**Comment by sokra** [comment]
I refactored it a bit...

**Comment by sokra** [comment]
Thanks

**Pull request merged**